src/reconcile/rec_write.c

```diff
@@ -976,8 +978,8 @@ __rec_destroy(WT_SESSION_IMPL *session, void *reconcilep)
 		return;
 	*(WT_RECONCILE **)reconcilep = NULL;
 
-	__wt_buf_free(session, &r->disk_image);
-	__wt_scr_free(session, &r->interim_buf);
+	__wt_buf_free(session, &r->disk_image[0]);
+	__wt_buf_free(session, &r->disk_image[1]);
```

Comment:
I'd steer clear of using scratch buffers (for pretty much anything, actually, but that's a bigger discussion).  Modern memory allocators like tcmalloc and jemalloc already stash recently freed memory for fast reuse.

I'd particularly avoid scratch buffers in reconciliation because having large chunks of memory hang around after eviction can be problematic when there are lots of active sessions.